src/core/src/op/space_to_depth.cpp

```diff
@@ -52,61 +52,48 @@ void ngraph::op::v0::SpaceToDepth::validate_and_infer_types() {
     set_output_type(0, get_input_element_type(0), output_shape);
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace {
-bool evaluate_space_to_depth(const HostTensorVector& outputs,
-                             const HostTensorVector& inputs,
-                             const std::size_t block_size,
-                             const ov::op::v0::SpaceToDepth::SpaceToDepthMode mode) {
+bool SpaceToDepth::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
+    OV_OP_SCOPE(v0_SpaceToDepth_evaluate);
+    OPENVINO_ASSERT(outputs.size() == 1);
+
     const auto& in = inputs[0];
     const auto& out = outputs[0];
-    size_t elem_size = in->get_element_type().size();
-
-    if (in->get_partial_shape().is_dynamic()) {
-        return false;
-    }
-
-    ov::reference::space_to_depth(in->get_data_ptr<char>(),
-                                  in->get_shape(),
-                                  out->get_data_ptr<char>(),
-                                  out->get_shape(),
-                                  block_size,
-                                  mode,
-                                  elem_size);
+    reference::space_to_depth(static_cast<const char*>(in.data()),
+                              in.get_shape(),
+                              static_cast<char*>(out.data()),
+                              out.get_shape(),
+                              m_blocksize,
+                              m_mode,
+                              in.get_element_type().size());
     return true;
 }
-}  // namespace
 
-bool ngraph::op::v0::SpaceToDepth::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
-    OV_OP_SCOPE(v0_SpaceToDepth_evaluate);
-    return evaluate_space_to_depth(outputs, inputs, m_blocksize, m_mode);
-}
-
-bool ngraph::op::v0::SpaceToDepth::has_evaluate() const {
+bool SpaceToDepth::has_evaluate() const {
     OV_OP_SCOPE(v0_SpaceToDepth_has_evaluate);
     return !get_input_partial_shape(0).is_dynamic();
```

Comment:
@jane-intel , @ilyachur 
Do you know why the input shape is checked here? I think this could return always true as evaluate not on element type and ov::Tensor cannot have dynamic shape.